File: OPT/TC3AutoDual/T3DualAuto/ADLINK/PCIS-DASK/Samples/.NET/VB.NET/7250/D7250_DIO/D7250_DIO/Form1.vb
Public Class Form1
    Dim m_dev As Short

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        m_dev = Register_Card(PCI_7250, 0)
        If (m_dev < 0) Then
            MessageBox.Show("Register_Card error!")
        End If
    End Sub

    Private Sub Form1_FormClosing(ByVal sender As System.Object, ByVal e As System.Windows.Forms.FormClosingEventArgs) Handles MyBase.FormClosing
        Dim err As Short
        If (m_dev >= 0) Then
            err = DASK.Release_Card(m_dev)
        End If
    End Sub

    Private Sub button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button1.Click
        Dim err As Short
        Dim out_value As Integer
        If (Int32.TryParse(textBox1.Text, out_value)) Then
            err = DO_WritePort(m_dev, 0, out_value)
            If (err < 0) Then
                MessageBox.Show("DO_WritePort error!")
            End If
        Else
            MessageBox.Show("Input error!")
        End If
    End Sub

    Private Sub button2_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles button2.Click
        Dim err As Short
        Dim int_value As UInteger
        err = DASK.DI_ReadPort(m_dev, 0, int_value)
        If (err < 0) Then
            MessageBox.Show("D2K_DI_ReadPort error!")
            Return
        End If
        textBox2.Text = int_value
    End Sub
End Class

File: OPT/TC3AutoDual/T3DualAuto/ADLINK/PCIS-DASK/Samples/.NET/C#/7250/D7250_DIO/D7250_DIO/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace D7250_DIO
{
    public partial class Form1 : Form
    {
        public short m_dev;
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            m_dev = DASK.Register_Card(DASK.PCI_7250, 0);
            if (m_dev < 0)
            {
                MessageBox.Show("Register_Card error!");
             
            }
        }

        private void Form1_FormClosing(object sender, FormClosingEventArgs e)
        {
            short ret;
            if (m_dev >= 0)
            {
                ret = DASK.Release_Card((ushort)m_dev);
            }
        }

        private void button1_Click(object sender, EventArgs e)
        {
            short ret;
            int out_value;
            if (Int32.TryParse(textBox1.Text, out out_value))
            {
                ret = DASK.DO_WritePort((ushort)m_dev, 0, (uint)out_value);
                if (ret < 0)
                {
                    MessageBox.Show("DO_WritePort error!");
                }
            }
            else
            {
                MessageBox.Show("Input error!");
            }
        }

        private void button2_Click(object sender, EventArgs e)
        {
            short ret;
            uint int_value;
            ret = DASK.DI_ReadPort((ushort)m_dev, 0, out int_value);
            if (ret < 0)
            {
                MessageBox.Show("D2K_DI_ReadPort error!");
                return;
            }
            textBox2.Text = String.Format("{0}", int_value);
        }

        private void label2_Click(object sender, EventArgs e)
        {

        }

        private void label1_Click(object sender, EventArgs e)
        {

        }

        private void textBox2_TextChanged(object sender, EventArgs e)
        {

        }

        private void textBox1_TextChanged(object sender, EventArgs e)
        {

        }
    }
}